### ISSUE TITLE
Document ActiveJob compatibility/configuration

### ISSUE THREAD

**Issue by bensheldon** [chanks/que#127]
I eagerly recommend Que to everyone, but the first question they ask is "Is it ActiveJob compatible?". The answer is Yes, but ActiveJob is not mentioned in the Readme nor is there an example in the docs directory. I'd be happy to submit a PR to improve the documentation for using Que with ActiveJob if this is an unintentional omission.

**Comment by chanks** [comment]
In the 1.0 branch I've actually removed everything that's Rails-specific (the Railtie, the migration generator, etc). Those parts have always been the most poorly tested and the most time consuming to maintain, especially since I don't personally use Que with Rails (My day job uses Rails + Sidekiq, and I have personal projects that use Que with Roda and Sinatra). I've also gotten frustrated with ActiveJob in the past, and I've grown kind of skeptical that a job queue is the right place for this kind of plug-and-play abstraction.

I've actually been considering whether it would be good to have a disclaimer on the Readme, saying that if the reader wants a quick and easy integration with Rails that they don't have to think too much about, they should probably be looking elsewhere. I'm happy if other people get use out of Que, and when I first launched it I was certainly hoping for a significant level of adoption, which is why I included the Rails integration in the first place. But these days I don't really feel like a large userbase for it is a goal I need to strive toward.

I think it'd be nice if somebody with more of an interest in the Rails stack maintained a gem that promised an easy integration with Que, and I'd certainly try to keep Que flexible enough to make their job easy. But I'm not really plugged into the Rails Way anymore, and I can't really justify the time expenditure to support everyone that is.

**Comment by chanks** [comment]
BTW, sorry if that reads as brusque. I don't mean to brush aside your concerns, I'm just laying out how I feel. Does my wanting to drop closer Rails support seem unreasonable to you, or anyone else reading?

**Comment by bensheldon** [comment]
@chanks that reads completely polite :smile: 

I might disagree that "a large userbase" and a small, focused, opinionated tool like Que are mutually exclusive. But I can agree and sympathize about externalizing the Rails integration into a separate gem if that's not where you want to spend time in development or support. As a middleground, you might consider still maintaining control (e.g. owning the repo and Rubygem) of a potential `que-rails` gem but offload the support to co-maintainers. Just my preference, but I'd rather have an officially blessed gem than multiple projects in various states of abandonment on Github. 

Do you have a public timeline/milestone for the 1.0 release? I hope there can be more community discussion and input; not to change your mind, but at least to pick up the work to be done.

**Comment by chanks** [comment]
Thanks. :smile: And yeah, I started a que-rails repository a while ago but stumbled around on how to test it. Having an official version that is mostly maintained by those more plugged into Rails sounds good to me.

I don't really have a timeline in mind for 1.0. I mean, the branch is public, but I don't know how much more I want to do with it. It's enough of a rewrite that I considered releasing it as a completely separate gem, but decided against it. At the very least I think there'll be a few betas (maybe even alphas?), so it won't be a sudden thing.

**Comment by CraigCottingham** [comment]
I just started switching from Sidekiq to Que for an application I'm working with, and the spotty integration with ActiveJob has been my only source of frustration. That being said, I can fully appreciate your not wanting to expend energy on a technology you don't use anymore. *That* being said, I'd still like to see Que work with ActiveJob, and am willing to help out with maintenance as needed.

**Comment by timdiggins** [comment]
Know this issue is quite old, but wondering what the current state of play/plan is -- I notice that there's an ActiveJob adapter for que in rails itself. (although there are arguments against using ActiveJob elsewhere: e.g. #156)

**Comment by timdiggins** [comment]
Starting to answer my own question: it looks like the 1.0 branch (in beta) seems to have rails and ActiveJob support fully baked-in (https://github.com/chanks/que/blob/v1.0.0.beta/CHANGELOG.md).

**Comment by chanks** [comment]
Yes, that's correct. You're welcome to use 1.0.0.beta, I'm using it in production at my day job. Bug reports welcome!

**Comment by timdiggins** [comment]
Great. Will try it out after some other changes in the codebase calm down.
